Transition InputBlurChange to use core Input component

npfitz opened the pull request:
PR transitions `InputBlurChange` to typescript and wrap the core `Input` component rather than a bare `input` element. The majority of the changes here are cosmetic and involved removing the `.input` class from many instances of `InputBlurChange` so that the input was rendered with double borders. The Specific Date filter was updated because while it already had an issue of horizontal scrolling in certain situations, the changes to `InputBlurChange` made adding a time impossible.

Main places that were impacted:
- Admin settings, especially when working with DB fields
- Date/Time filters
- Viz Colum Formatting and Series settings
- Click Behavior Custom URL

npfitz committed:
Convert to TypeScript, handle ColumnItem and MetadataTable

npfitz committed:
more progress

npfitz committed:
handled numeric inputs

npfitz committed:
Transition InputBlurChange to core Input component

npfitz committed:
Removing input class from legacy date picker just incase

alxnddr commented on frontend/src/metabase/components/InputBlurChange.tsx:
```diff
@@ -0,0 +1,65 @@
+ * `onBlurChange` feature, otherwise you should use <input> directly
+ */
+
+interface InputBlurChangeProps {
```
Wdyt about changing it like this?
```ts
import Input, { InputProps } from "metabase/core/components/Input";
// ...
interface InputBlurChangeProps extends InputProps {
  onBlurChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
}
```

npfitz replied:
I had this thought earlier but clearly forgot to do it. Yes, much cleaner 👍

alxnddr commented on frontend/src/metabase/admin/datamodel/components/database/MetadataTable/MetadataTable.styled.tsx:
```diff
@@ -41,20 +41,31 @@ const headerInputsStyles = css`
-  font-size: 20px;
-  color: ${color("text-dark")};
-  border-radius: 8px 8px 0 0;
+  //${headerInputsStyles}
```
Looks like you forgot the commented styles

alxnddr commented on frontend/src/metabase/admin/datamodel/components/database/MetadataTable/MetadataTable.styled.tsx:
```diff
@@ -41,20 +41,31 @@ const headerInputsStyles = css`
-  border-radius: 0 0 8px 8px;
-  font-weight: 400;
-  font-size: 14px;
+  //${headerInputsStyles}
```
Here too

alxnddr commented on frontend/src/metabase/admin/datamodel/components/database/ColumnItem/ColumnItem.jsx:
```diff
@@ -59,7 +59,7 @@ class Column extends Component {
                 variant="primary"
                 style={{ minWidth: 420 }}
-                className="AdminInput TableEditor-field-name float-left inline-block rounded text-bold"
+                className="TableEditor-field-name float-left inline-block rounded text-bold"
```
Looks like we can remove `TableEditor-field-name rounded text-bold` here and below

alxnddr commented on frontend/src/metabase/query_builder/components/filters/pickers/DatePicker/SpecificDatePicker.styled.tsx:
```diff
@@ -31,9 +30,9 @@ export const DateInputContainer = styled.div<DateInputContainerProps>`
-  border: 1px solid
-    ${({ isActive }) => (isActive ? color("brand") : color("border"))};
-  border-radius: 0.5rem;
+  // border: 1px solid
```
Looks like we can remove this

alxnddr reviewed (commented):
I believe we need to do a little extra and make them visually fit into UI nicely as before

I noticed the input started being too tall in the viz settings. We previously had an issue that the sizes of select buttons and inputs were different:
<img width="328" alt="Screen Shot 2022-12-09 at 8 28 24 PM" src="https://user-images.githubusercontent.com/14301985/206811570-ef1bf8f7-a1bc-4304-ae47-1d5581ae8bdb.png">

The same issue is in the Data Model:
<img width="1374" alt="Screen Shot 2022-12-09 at 8 27 56 PM" src="https://user-images.githubusercontent.com/14301985/206811527-d2fbbd81-8369-4bf7-ad73-9036886242e7.png">

Previously, using `size=small` would probably solve the issue, but the small variant got smaller:

<img width="215" alt="Screen Shot 2022-12-09 at 8 25 09 PM" src="https://user-images.githubusercontent.com/14301985/206811777-1fc64028-263e-4013-b849-ab293f22bd40.png">
<img width="192" alt="Screen Shot 2022-12-09 at 8 26 09 PM" src="https://user-images.githubusercontent.com/14301985/206811784-3f042d1a-e258-4b14-8d5f-3cd6c8e04ec9.png">

npfitz commented:
@alxnddr I updated the areas you mentioned by using `inputTypography` and `inputPadding` from core styles. But there might be an arguement for having those be the defaults in the core component itself since there are probably lots of places we're overriding these values

npfitz committed:
Small style adjustments in Admin and Viz Settings

npfitz committed:
working on tests

ranquild commented on frontend/src/metabase/components/NumericInput.jsx:
```diff
@@ -1,10 +1,9 @@
 
 const NumericInput = ({ value, onChange, ...props }) => (
-  <InputBlurChange
+  <NumbericImputBlurChange
```
Typo, iMput

ranquild commented on frontend/src/metabase/components/NumericInput.styled.tsx:
```diff
@@ -0,0 +1,9 @@
+import InputBlurChange from "./InputBlurChange";
+
+export const NumbericImputBlurChange = styled(InputBlurChange)`
+  ${InputBlurChange.Field} {
```
There is `fullWidth` prop available on `Input`. Would it be enough?

npfitz replied:
The problem is that fullWidth applies `width: 100%` to both the `root` and the `field`, making it hard difficult to layout/size without another container that is sized specifically.

npfitz replied:
Talked with @ranquild. Going to try setting on the usingInput fullWidth by default, and styling the component as needed across the app

ranquild commented on frontend/src/metabase/components/InputBlurChange.tsx:
```diff
@@ -0,0 +1,53 @@
+    }
+  };
+
+  const inputProps = _.omit(props, "onBlur");
```
btw would it be better to call the original `onBlur` in `handleBlur` rather than omit it?

ranquild commented on frontend/src/metabase/components/InputBlurChange.tsx:
```diff
@@ -0,0 +1,53 @@
+    setInternalValue(value);
+  }, [value]);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```
useCallback? it should prevent rerenders in common cases

npfitz replied:
Oh right. I'll update that!

npfitz committed:
Tests passing

npfitz committed:
Adjusting styling

npfitz commented on frontend/src/metabase/admin/datamodel/components/database/MetadataTable/MetadataTable.styled.tsx:
```diff
@@ -41,20 +42,25 @@ const headerInputsStyles = css`
-  font-size: 20px;
-  color: ${color("text-dark")};
-  border-radius: 8px 8px 0 0;
+  ${Input.Field} {
```
This is a very specific instance where the field is styled heavily

npfitz commented on frontend/src/metabase/components/InputWithSelectPrefix.styled.tsx:
```diff
@@ -0,0 +1,11 @@
+  width: auto;
+
+  ${Input.Field} {
+    border: none;
```
Another field that is heavily styled

npfitz committed:
PR Cleanup

alxnddr commented:
@npfitz it seems like we still need a large variant in some contexts such as collection/dashboard forms because they have text areas with larger padding and font. I created a tiny PR with a few tweaks, can you please have a look: https://github.com/metabase/metabase/pull/27217

npfitz commented:
Certainly! Ill give it a review in the morning

alxnddr reviewed (approved):
Let's merge https://github.com/metabase/metabase/pull/27217 into this branch, and then merge it into `master`

qnkhuat committed:
Suppress console log when using `with-log-messages-for-level` (#26468)

* disable additivity when using "with-log-level"

* only set when parent is root

* wording

* remove a test that is no longer needed

* fix indents

* one missing indent fix

nemanjaglumac committed:
[CI] Add `concurrency` to the `drivers` workflow (#27223)

calherries committed:
Fix datetime-diff helper text (#27224)

calherries committed:
Improve datetimeDiff type error message (#27225)

* Update test

* Update implementations

dpsutton committed:
Relieve db pressure on api/health check (#27192)

* Relieve db pressure on api/health check

https://github.com/metabase/metabase/issues/26266

Servers under heavy load can be slow to respond to the api/health
check. This can lead to k8s killing healthy instances happily humming
along serving requests.

One idea floated was to use QoSFilters
https://www.eclipse.org/jetty/javadoc/jetty-9/org/eclipse/jetty/servlets/QoSFilter.html
to prioritize those requests in front of others. But I suspect this
might not be our bottleneck.

Our health endpoint was updated to see if it could acquire an endpoint
when we were dealing with connection pool issues. We were reporting the
instance was healthy once it has finished the init process, but would
report healthy if 60/15 app-db connections were used and no actual
queries could complete.

The remedy was adding
`(sql-jdbc.conn/can-connect-with-spec? {:datasource (mdb.connection/data-source)})`
to the endpoint. But now to get information about the health of the
system we have to wait in the queue to get a datasource.

The hope is that this change which monitors for recent db
checkins (query success) and checkouts (query begun) can be a proxy for
db activity without having to wait for a connection and hit the db ourselves.

Some simple and crude benchmarking:
- use `siege` to hit `api/database/<app-db>/sync_schema`
- in a separate tab, use `siege` to hit `api/health`

Three trials with unconditional db access and conditional db
access (look for recent activity set by the new `ConnectionCustomizer`).

One siege client is synching the app-db's schema with 80 clients each
sending 60 requests. the other has 1 client sending 60 requests to api/health.

Run             |  Elapsed Time | max tx  | tx rate
 before change  |    7.16s      |  0.79s  |  8.38 tx/s
 before change  |   23.91s      |  1.44s  |  2.51 tx/s
 before change  |   13.00s      |  0.50s  |  4.62 tx/s
----------------------------------------------------
 after change   |    4.46s      |  0.27s  |  13.45 tx/s
 after change   |    5.81s      |  0.61s  |  10.33 tx/s
 after change   |    4.54s      |  0.44s  |  13.22 tx/s

Full(er) results below:

```
Unconditional db access
=======================

siege -c80 -r 40 "http://localhost:3000/api/database/2/sync_schema POST" -H "Cookie: $SESSION"

siege -c 1 -r 60 "http://localhost:3000/api/health"

Elapsed time:		        7.16 secs
Response time:		        0.12 secs
Transaction rate:	        8.38 trans/sec
Longest transaction:	        0.79
Shortest transaction:	        0.01

Elapsed time:		       23.91 secs
Response time:		        0.40 secs
Transaction rate:	        2.51 trans/sec
Longest transaction:	        1.44
Shortest transaction:	        0.02

Elapsed time:		       13.00 secs
Response time:		        0.22 secs
Transaction rate:	        4.62 trans/sec
Longest transaction:	        0.50
Shortest transaction:	        0.06

Conditional db access
==============================================================

Elapsed time:		        4.46 secs
Response time:		        0.07 secs
Transaction rate:	       13.45 trans/sec
Longest transaction:	        0.27
Shortest transaction:	        0.01

Elapsed time:		        5.81 secs
Response time:		        0.10 secs
Transaction rate:	       10.33 trans/sec
Longest transaction:	        0.61
Shortest transaction:	        0.00

Elapsed time:		        4.54 secs
Response time:		        0.08 secs
Transaction rate:	       13.22 trans/sec
Longest transaction:	        0.44
Shortest transaction:	        0.01
```

* Remove reflection in `.put` call (not the reflections trategy)

also remove the call to `classloader/the-classloader` as it did nothing

* Comment and settle on a single method

* tests

* select from db twice

had a failure in CI. give it time to do its thing with another db call

* block to wait for timestamp update?

* unflake the tests

tasks and events from outside the thread can hit the db. the
ConnectionCustomizer is also run from c3p0 controlled threads so we
can't easily isolate everything to our thread

Was running

```clojure
(comment
  (dotimes [n 5]
    (dotimes [_ 100]
      (recent-activity-test)
      (CheckinTracker-test))
    (println (* (inc n) 100)))
    )
```

to run the tests 500 times and would keep getting flakes at a rate
~1/100 to 1/500. Just frustration for the future.

* typehint

* Switch it up a bit

Tests were flaking in h2 and I don't know why. I'm switching to just
updating recent activity on most methods.

alxnddr committed:
add input size=large (#27217)

npfitz committed:
final touches
